docs/solutions/windows/scripts/fleetd-healthcheck.ps1

```
@@ -0,0 +1,378 @@
+}
+
+# Resource exhaustion (low memory) - Windows' rough equivalent of the Linux OOM killer
+try {
+  $ResourceEvents = Get-WinEvent -FilterHashtable @{ LogName = 'System'; ProviderName = 'Microsoft-Windows-Resource-Exhaustion-Detector'; StartTime = $Since } -ErrorAction SilentlyContinue
+  if ($ResourceEvents) {
+    Warn "Low-memory/resource-exhaustion activity detected in last ${LookbackHours}h"
+    $ResourceEvents | Select-Object TimeCreated, Id, Message | Export-Csv -Path (Join-Path $SystemEventsDir 'resource_exhaustion_events.csv') -NoTypeInformation
+  }
+} catch {
+  # Provider may not be present on all SKUs - not a failure condition
+}
```

PSScriptAnalyzer (1.25.0)

[warning] Empty catch block is used. Please use Write-Error or throw statements in catch blocks.

(PSAvoidUsingEmptyCatchBlock)